module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -491,43 +514,37 @@ namespace module::platform::darwin {
                                              config.motionFilter.noise.measurement.flatFootOrientation,
                                              MeasurementType::FLAT_FOOT_ORIENTATION());
                     }
+                    // Else the foot is off the ground, so we make sure that for the next time step, we know that
+                    // this time step, the foot was off the ground
                     else if (!foot_down) {
                         previous_foot_down[side] = false;
                     }
 
+                    // Note that the Hwf is set, even if the foot is not down. This means that moving feet in the air
+                    // will have an Hwf associated with them which is the transform from when that foot last hit
+                    // the ground
                     sensors->feet[side].Hwf = footlanding_Hwf[side].matrix();
                 }
 
                 // Calculate our time offset from the last read
-                double deltaT = std::max(
+                // too hard to read
+                const double deltaT = std::max(
                     (input.timestamp - (previousSensors ? previousSensors->timestamp : input.timestamp)).count()
                         / double(NUClear::clock::period::den),
                     0.0);
 
                 // Time update
                 motionFilter.time(deltaT);
 
-                // Gives us the quaternion representation
+                // Convert the motion filter's state vector to a nicer representation, so we can access its elements
                 const auto o = MotionModel<double>::StateVec(motionFilter.get());
 
-                // Map from world to torso coordinates (Rtw)
+                // We make Hwt first, because `o` is in world space
                 Eigen::Affine3d Hwt;
                 Hwt.linear()      = o.Rwt.toRotationMatrix();
-                Hwt.translation() = o.rTWw;
+                Hwt.translation() = -o.rTWw;
```

Comment:
pretty sure this is negative to what it should be in master. On master we are storing a vector from world to torso in a transform from torso to world. That seems wrong to me

Reply:
`Hwt.translation()` should always be `rTWw` by definition. `-rTWw` would make it `rWTw` which is definitely wrong here

Reply:
https://nubook.nubots.net/system/foundations/mathematics#homogeneous-transformations 😄

Reply:
yikes. I had this wrong then lol 😬

Reply:
The way I remember it is if you have `Hab`, it transforms from `b` to `a`, however, what it is constructed from is 4 vectors as the columns, all these values are measured in `a` space. They are the X, Y and Z axis of the `b` space as measured in `a` space, and then the translation vector to `b` space measured in `a` space.

Reply:
oh, so the rotation matrix is like a change of basis. Of course it is... I never realised that lol :facepalm: 
thanks!

Reply:
It's literally a change of basis :P